### PR TITLE
[Request] Add Slack actions to notifications

### PR DESCRIPTION
**Pull request by AlexLardschneider**
This PR adds the Slack action parameters according to the Slack documentation available at https://api.slack.com/docs/message-buttons.

The action buttons can be added as follows:

```yaml 
actions:
  - type: "button"
    text: 'Test'
    url: 'http://example.com'
    style: 'danger'
```

### CODE REVIEW

File: config/notifiers.go [hunk 1]
```diff
@@ -220,6 +220,34 @@ func (c *PagerdutyConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 	return nil
 }
 
+// SlackAction configures a single Slack action that is sent with each notification.
+// Each action must contain a type, text, and url.
+// See https://api.slack.com/docs/message-attachments#actions for more information.
```
Comment (simonpasquier) [inline config/notifiers.go]: The correct link is `https://api.slack.com/docs/message-attachments#action_fields`.

File: config/notifiers.go [hunk 2]
```diff
@@ -220,6 +220,34 @@ func (c *PagerdutyConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
...
+type SlackAction struct {
+	Type	string 	`yaml:"type,omitempty" json:"type,omitempty"`
+	Text	string 	`yaml:"text,omitempty" json:"text,omitempty"`
+	Url   string 	`yaml:"url,omitempty" json:"url,omitempty"`
+	Style string 	`yaml:"style,omitempty" json:"style,omitempty"`
```
Comment (simonpasquier) [inline config/notifiers.go]: Please align the `string` identifiers.

**Comment by AlexLardschneider** [comment]
Not sure how to sign the old commits. Can you point me in the right direction on how to do that? Will fix the other issues.

**Comment by simonpasquier** [comment]
> Not sure how to sign the old commits.

You can do `git commit --amend <commit SHA1>` to edit the unsigned commits. Then `git push --force-with-lease`.

**Comment by AlexLardschneider** [comment]
Commits should be signed now.

**Comment by stuartnelson3** [comment]
Thanks for the PR! Sorry for the long delay, I'll look at the code ASAP

File: config/notifiers.go [hunk 3]
```diff
@@ -220,6 +220,34 @@ func (c *PagerdutyConfig) UnmarshalYAML(unmarshal func(interface{}) error) error
 	return nil
 }
 
+// SlackAction configures a single Slack action that is sent with each notification.
+// Each action must contain a type, text, and url.
+// See https://api.slack.com/docs/message-attachments#action_fields for more information.
+type SlackAction struct {
+  Type  string  `yaml:"type,omitempty" json:"type,omitempty"`
+  Text  string  `yaml:"text,omitempty" json:"text,omitempty"`
+  Url   string  `yaml:"url,omitempty" json:"url,omitempty"`
```
Comment (stuartnelson3) [inline config/notifiers.go]: this should be `URL` (go style thing)

**Comment by AlexLardschneider** [comment]
Made the requested changes, can you please check?

**Pull request merged**